assets/blocks/quiz/question-categories.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Get the question categories.
+ *
+ * @return {Object[]} Term objects.
+ */
+export const useQuestionCategories = () => {
+	const questionCategories = useSelect( ( select ) => {
+		const terms = select( 'core' ).getEntityRecords(
+			'taxonomy',
+			'question-category',
+			{
+				per_page: -1,
+			}
+		);
+
+		if ( terms && terms.length ) {
+			return terms.map( ( term ) => ( {
+				...term,
+				name: unescape( term.name ),
+			} ) );
+		}
+
+		return terms;
+	} );
+
+	const getById = ( termId ) => {
```

Comment:
For now, I think it's okay, but I wonder about cases when we use the getById inside loops. We could refactor to make an object by id using the lodash `keyBy` (and maybe memorize it based on the `questionCategories`?), so we could just return the object property through the `getById` function instead `find` it.

But really not critical for now.

Reply:
On digging into this, I realized there was some inefficiency in doing the unescape of the title in the `useSelect`. Instead, I memoized that as well and now it is doing much less work. 7e4df24922caf47ed2354a7ae532ba42df764b63